src/background.js

```diff
@@ -469,33 +445,14 @@ const rollout = {
         }
       }
     });
-  },
 
-  async main() {
-    // Listen to the captive portal when it unlocks
-    browser.captivePortal.onStateChanged.addListener(rollout.onReady);
-
-    // If the captive portal is already unlocked or doesn't exist,
-    // run the measurement
-    let captiveState = await browser.captivePortal.getState();
-    log("Captive state:", captiveState);
-    if ((captiveState === "unlocked_portal") ||
-        (captiveState === "not_captive")) {
-      await rollout.onReady({state: captiveState});
-    }
+    browser.captivePortal.onConnectivityAvailable.addListener(
+      rollout.runStartupHeuristics()
```

Comment:
The return value of this call (which is a promise), is passed to `addListener`, which expects a function.
You either want to wrap this call in an anonymous function, or pass the function handle without calling it.
I recommend the former, because if you just pass the function handle, the function's scope will not have access to the same `this` which can lead to bugs (you'd need to call bind() to make that work)